lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2504,6 +2503,33 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     return Walker.visitedOriginallyDefinedIn;
   }
 
+  /// Returns true if the type contains an imported C++ type. Due to
+  /// various unimplemented features these cannot round-trip through
+  /// the ASTDemangler.
+  bool containsCxxType(Type T) {
+    return T.findIf([&](Type t) -> bool {
+      if (auto *decl = t->getAnyNominal()) {
+        if (auto *clangDecl = decl->getClangDecl()) {
+          if (isa<clang::CXXRecordDecl>(clangDecl))
+            return true;
+
+          if (isa<clang::ClassTemplateSpecializationDecl>(clangDecl))
+            return true;
+
+          auto *dc = clangDecl->getDeclContext();
+          while (!isa<clang::TranslationUnitDecl>(dc)) {
+            if (isa<clang::NamespaceDecl>(dc))
+              return true;
+
+            dc = dc->getParent();
+          }
```

Comment:
What should be the expected result for an enum class declared inside of a C++ template class?